src/main/kotlin/tay/example/manage_stock_of_book/service/BookService.kt

```diff
@@ -0,0 +1,161 @@
+package tay.example.manage_stock_of_book.service
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.data.domain.Page
+import org.springframework.data.domain.PageRequest
+import org.springframework.http.ResponseEntity
+import org.springframework.stereotype.Component
+import tay.example.manage_stock_of_book.api.exception.IDNotFoundException
+import tay.example.manage_stock_of_book.api.request.CreateBookDTO
+import tay.example.manage_stock_of_book.api.request.UpdateBookDTO
+import tay.example.manage_stock_of_book.model.Book
+import tay.example.manage_stock_of_book.model.Category
+import tay.example.manage_stock_of_book.repository.BookRepository
+import tay.example.manage_stock_of_book.repository.CategoryRepository
+import java.lang.RuntimeException
+import java.net.IDN
+
+
+@Component
+class BookService (
+        @Autowired
+        var bookRepository: BookRepository,
+
+        @Autowired
+        var categoryRepository: CategoryRepository
+){
+        var books: Page<Book>? = null
+        var category: Category? = null
+
+        fun getListBook(value: String = "null", categoryId: Long = 0, page: Int = 0, limit: Int = 0): ResponseEntity<Any> {
```

Comment:
Please refactor this method function getListBook
- No need to have many condition to filter all the params is null or not
- Make sure there are only 2 condition to filter params is null or not